packages/react-dom/src/__tests__/ReactDOMComponentTree-test.js

```diff
@@ -10,32 +10,42 @@
 'use strict';
 
 describe('ReactDOMComponentTree', () => {
-  var React;
-  var ReactDOM;
-  var ReactDOMComponentTree;
-  var ReactDOMServer;
+  let React;
+  let ReactDOM;
+  let ReactDOMServer;
 
   function renderMarkupIntoDocument(elt) {
-    var container = document.createElement('div');
+    const container = document.createElement('div');
     // Force server-rendering path:
     container.innerHTML = ReactDOMServer.renderToString(elt);
     return ReactDOM.hydrate(elt, container);
   }
 
-  function getTypeOf(instance) {
-    return instance.type;
+  function simulateInput(elem, value) {
+    const inputEvent = new Event('input', {
+      bubbles: true,
+    });
+    setUntrackedInputValue.call(elem, value);
+    elem.dispatchEvent(inputEvent);
   }
 
-  function getTextOf(instance) {
-    return instance.memoizedProps;
+  function simulateClick(elem) {
+    const event = new MouseEvent('click', {
+      bubbles: true,
+    });
+    elem.dispatchEvent(event);
   }
 
+  const setUntrackedInputValue = Object.getOwnPropertyDescriptor(
+    HTMLInputElement.prototype,
+    'value',
+  ).set;
+
   beforeEach(() => {
     React = require('react');
     ReactDOM = require('react-dom');
-    // TODO: can we express this test with only public API?
-    ReactDOMComponentTree = require('../client/ReactDOMComponentTree');
     ReactDOMServer = require('react-dom/server');
+    document.innerHTML = '';
```

Comment:
How do you feel about following this approach instead? Feels a tiny but more obvious what's going on and why IMO. https://github.com/facebook/react/pull/11517/commits/d7745d69917b28298d27b8e411329ca67cae4668

Reply:
I like it. Tests are more focussed on testing and we move set up to where it should be - thanks for the pointer. I'll adopt this approach.